gossip/types.go

```diff
@@ -0,0 +1,14 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+const HashLength = 8
+
+type Hash [HashLength]byte
+
+func HashFromBytes(b []byte) Hash {
```

Comment:
I've always tried to avoid these functions (while they are super nice to use) they kind if hide bugs if `b` isn't formatted correctly. Which is why the `ids.ToID` function in avalanchego returns an error if the length of the slice isn't correct

Reply:
Got rid of this utility function, instead we're manually copying `ids.ID` and `transaction.Hash` into `gossip.Hash` (which is safe because they are all the same size), and in the `gossip.Handler` we do a length check and error if it doesn't match the expected length.